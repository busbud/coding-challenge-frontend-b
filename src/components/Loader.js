import React from 'react'

class Loader extends React.Component {
  render(){
    return(
      <div className='loader'>
        <div className='loading'>
          <div className='loading-bar'></div>
          <div className='loading-bar'></div>
          <div className='loading-bar'></div>
          <div className='loading-bar'></div>
        </div>
      </ div>
    )
  }
}

export default Loader
