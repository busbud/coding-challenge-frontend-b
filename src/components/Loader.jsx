import React from 'react';
import createReactClass from 'create-react-class';
import { CircularProgress } from 'material-ui';


const Loader = createReactClass({
  getInitialState() {
		return {
      height: (window.innerHeight/2)-90
		};
  },
  resize: function() {
      this.setState({height: (window.innerHeight/2)-90});
  },
  componentWillMount: function() {
      this.resize();
  },
  componentDidMount: function() {
      window.addEventListener('resize', this.resize);
  },
  componentWillUnmount: function() {
      window.removeEventListener('resize', this.resize);
  },
  render(){
    return (
      <div className='loader-overlay'>
        <CircularProgress style={{marginTop:this.state.height}} size={60} thickness={7} />
      </div>
    );
  }
});

export default Loader;
