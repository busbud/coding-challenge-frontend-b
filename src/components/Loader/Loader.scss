@import "../../vars";
@import "../../susy";

.loader {

    @include span(10);
    @include push(1);

    &__animation {
        width: 40px;
        height: 40px;
        margin: 100px auto;
        background-color: white;

        border-radius: 100%;  
        animation: sk-scaleout 1.0s infinite ease-in-out;
    }
}

@keyframes sk-scaleout {
  0%   {  transform: scale(0);  } 
  100% {  transform: scale(1.0);
          opacity: 0; }
}