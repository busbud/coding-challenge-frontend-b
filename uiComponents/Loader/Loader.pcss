.__SCOPE {
  align-items: center;
  display: flex;
  flex-direction: column;
  justify-content: center;
  pointer-events: none;
  width: 100%;
  height: 100%;

  > svg {
    display: block;
    margin-bottom: 5px;
  }

  > p {
    color: #000;
    font-size: 12px;
  }
}
