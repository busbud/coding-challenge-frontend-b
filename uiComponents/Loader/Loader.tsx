import React from 'react';
import Styled from 'react-styles-injector';

import styles from './Loader.pcss';

export interface ILoaderProps {
  className?: string;
  color?: string;
  label?: string;
  style?: React.CSSProperties;
  width?: number;
}

export const Loader = ({
  className,
  color,
  label,
  style,
  width,
}: ILoaderProps): JSX.Element => {
  return (
    <Styled styles={styles} style={style} className={className || ''}>
      <svg
        height={`${width || '40'}px`}
        viewBox="0 0 50 50"
        width={`${width || '40'}px`}
        xmlns="http://www.w3.org/2000/svg">
        <path
          fill={color || '#000'}
          d="M25.251,6.461c-10.318,0-18.683,8.365-18.683,18.683h4.068c0-8.071,6.543-14.615,14.615-14.615V6.461z">
          <animateTransform
            attributeType="xml"
            attributeName="transform"
            type="rotate"
            from="0 25 25"
            to="360 25 25"
            dur="0.5s"
            repeatCount="indefinite"
          />
        </path>
      </svg>

      {label && <p style={{ ...(color && { color }) }}>{label}</p>}
    </Styled>
  );
};
