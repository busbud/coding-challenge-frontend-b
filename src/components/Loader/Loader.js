import React, { PropTypes } from 'react';
import './Loader.scss';

class Loader extends React.Component {
    constructor(props) {
        super(props);
    }
    render() {

        return (
            <div className="loader">
                <div className="loader__animation"></div>
            </div>
        );
    }
}

export default Loader;