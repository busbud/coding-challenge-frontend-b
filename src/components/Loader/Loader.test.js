// third-party libraries
import React from 'react';
import { shallow } from 'enzyme';

// components
import Loader from './index'

describe('NavBar Component', () => {
  it('should be rendered properly', () => {

    const wrapper = shallow(<Loader />);
    expect(wrapper).toMatchSnapshot();
  });
})
