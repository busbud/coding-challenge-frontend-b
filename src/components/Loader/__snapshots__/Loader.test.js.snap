// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar Component should be rendered properly 1`] = `ShallowWrapper {}`;
